android/src/main/java/nordpol/ui/NfcGuideView.java

```diff
@@ -0,0 +1,276 @@
+package nordpol.android;
+
+import android.content.res.TypedArray;
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewTreeObserver;
+import android.view.animation.AccelerateDecelerateInterpolator;
+import android.view.animation.AccelerateInterpolator;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
+
+import nordpol.android.R;
+
+/**
+ * A custom view to help users position NFC device as well as keep it around
+ * while transferring. Using the full flow with three states (positioning,
+ * transaction and finished-done or finished-fail) is recommended for longer transactions. For
+ * shorter transactions (such as UID lookup) only the first state should be
+ * used.
+ *
+ * Usage: Declare the View and call the different states directly on it. The
+ * available states are declared in the enum NfcGuideViewStatus and are called
+ * with the method setCurrentStatus. The NFC device graphic shown to the user
+ * can be changed. This is currently only possible with the XML attribute
+ * nfc_device. The "graphic-states" available for the attribute are declared in
+ * the android/src/main/res/values/attrs.xml file.
+ */
+public class NfcGuideView extends RelativeLayout {
+    private static final int GUIDE_ITEMS_START_DISTANCE = 100;
+    private static final int ANIMATION_DURATION_SHORT = 100;
+    private static final int ANIMATION_DURATION_MEDIUM = 200;
+    private static final int VALUE_ANIMATOR_DEFAULT_START_DELAY = 0;
+    private final int START_DELAY_LONG = 500;
+    private int mNfcGuidePhoneWidth = 0;
```

Comment:
I would group these ones with the other member variables below, rather than with the static defines up here.